File: mycollab-caching-premium/src/main/java/com/esofthead/mycollab/cache/service/CleanCacheCommand.java
package com.esofthead.mycollab.cache.service;

import com.esofthead.mycollab.cache.CleanCacheEvent;
import com.esofthead.mycollab.module.GenericCommand;
import com.google.common.eventbus.AllowConcurrentEvents;
import com.google.common.eventbus.Subscribe;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

/**
 * @author MyCollab Ltd
 * @since 5.1.3
 */
@Component
public class CleanCacheCommand extends GenericCommand {

    @Autowired
    private CacheService cacheService;

    @AllowConcurrentEvents
    @Subscribe
    public void cleanCaches(CleanCacheEvent event) {
        cacheService.removeCacheItems(event.sAccountId().toString(), event.cls());
    }
}

File: mycollab-esb/src/main/scala/com/mycollab/module/common/esb/CleanCacheCommand.scala
package com.mycollab.module.common.esb

import com.google.common.eventbus.{AllowConcurrentEvents, Subscribe}
import com.mycollab.cache.CleanCacheEvent
import com.mycollab.cache.service.CacheService
import com.mycollab.module.esb.GenericCommand
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component

/**
  * @author MyCollab Ltd
  * @since 5.3.5
  */
@Component
class CleanCacheCommand extends GenericCommand {
  @Autowired private val cacheService: CacheService = null
  
  @AllowConcurrentEvents
  @Subscribe def cleanCaches(event: CleanCacheEvent) {
    cacheService.removeCacheItems(event.sAccountId.toString, event.cls: _*)
  }
}
